resources/default-form-data/custom-rule/CustomRuleRector.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Cast\String_;
+use PhpParser\Node\Scalar\LNumber;
+use Rector\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+final class CustomRuleRector extends AbstractRector
+{
+    public function getNodeTypes(): array
+    {
+        return [String_::class];
+    }
+
+    /**
+     * @param String_ $node
+     */
+    public function refactor(Node $node)
+    {
+        // change to number
+        return new LNumber(100);
```

Comment:
is the page allow user to try their own rule? beware of user add dangerous func call, eg: `shell_exec`, `system`, etc.

this was the incident because of load rule config at demo page that user uses `shell_exec` func call, that's why the rule is added to not allow add func call on config box:

- https://github.com/rectorphp/getrector-com/pull/1967


![Screenshot 2024-06-24 at 19 37 40](https://github.com/rectorphp/getrector-com/assets/459648/e1382431-f8f1-4d0a-ab18-64b23716a620)

Reply:
Yes, so people share and try rules easily.

The rule content is evaluated the same way Rector demo config is. So all the forbidden methods are checked here too. Do you think that's good enough?

Reply:
Rector box config rule currently limit all `FuncCall_` for safest belt, which if we apply on this page, may make create custom rule hard to create as user may rely on native PHP function.

We probably needs list of "safe" functions :

- utilize `PureFunctionDetector::detect()` from Rector for complete list.
- array functions (side effect only on current array, eg: array_pop, etc)
- etc

That needs to be test carefully locally test first before publish online, probably mark this page "dev" only first to test locally.